### ISSUE TITLE
CA1508 False positive on using statement

### ISSUE THREAD

**Issue by mayerj** [dotnet/roslyn-analyzers#2985]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v2.9.6 (Latest)

#### Diagnostic ID

CA1508 - Avoid Dead Conditional Code

#### Repro steps

![image](https://user-images.githubusercontent.com/4032461/67692213-38729000-f976-11e9-8d79-c647443cdd0e.png)

```
		static void Test(byte[] data)
		{
			using (var ms = new MemoryStream(data))
			{
				Console.WriteLine(ms);
			}
		}

		static void TestError(byte[] data)
		{
			using (var ms = new MemoryStream(data))
			{
				Console.WriteLine(bool.Parse(ms.ToString()));
			}
		}
```

#### Expected behavior

No CA1508 on the "new MemoryStream(data)" part

It's not clear what the rule wants me to remove.  I suspect it should be ignoring the generated check added by the using block.

Note that this only seems to happen if you pass the using object to another function, or deference it?  The Test(...) method doesn't show the error.

#### Actual behavior

CA1508

**Comment by bkoelman** [comment]
I'm still getting this false positive today (VS 2019 16.10 in .NET5 project with `AnalysisMode:AllEnabledByDefault`) when no variable is assigned:

```c#
using (new CodeTimer("Recalculate rankings"))
```
